### DISCUSSION TITLE
Orca Confusion

### DISCUSSION THREAD

**Discussion by gpena208777** [xblax/flashforge_ad5m_klipper_mod#79] [General]
Hi all
First of all great work. 

When updating the Orca Machine Start g code and stop g code, it references this for the start and stop.

Am I understanding correctly that all I do is paste that into this Orca screen(see image)
<img width="770" alt="klipper" src="https://github.com/xblax/flashforge_adm5_klipper_mod/assets/155690343/3d907d40-977d-4ca2-b05d-db6d3044785f">

Thanks

```
    {% set bed_temp = params.BED_TEMP|default(60)|float %}             ; bed temp, usually set by slicer
    {% set extruder_temp = params.EXTRUDER_TEMP|default(200)|float %}  ; extruder temp, usually set by slicer
    {% set disable_priming = params.DISABLE_PRIMING|default(false) %}  ; allow disabling of priming before print
    {% set force_leveling = params.FORCE_LEVELING|default(false) %}    ; allow disabling of priming before print

    SET_FAN_SPEED FAN=pcb_fan SPEED=1
    M140 S{bed_temp}        ; start bed heating
    SET_GCODE_OFFSET Z=0.0  ; reset the G-Code Z offset (adjust Z offset if needed)
    G28                     ; home the printer

    # auto leveling, if no mesh loaded or force_leveling was set
    {%if (not printer['bed_mesh'].profile_name) or force_leveling %}
        AUTO_BED_LEVEL BED_TEMP={bed_temp} EXTRUDER_TEMP={extruder_temp}
    {% endif %}

    G90                     ; use absolute coordinates
    G1 Z5 F1800             ; move the nozzle near the bed
    # wait for bed to reach temp, don't wait for stabilization
    TEMPERATURE_WAIT SENSOR=heater_bed MINIMUM={bed_temp}
    G1 Z0.15 F300           ; move the nozzle very close to the bed
    M109 S{extruder_temp}   ; set and wait for nozzle to reach temperature
    G1 Z5 F1800             ; raise z before first move

    # prime nozzle before print if not disabled
    {% if not disable_priming %}
        _PRIME_NOZZLE
    {% endif %}
```

And this for the stop 

```
    G91                     ; relative positioning
    G1 X-2 Y-2 E-5 F2000    ; move away while retracting filament
    _COMMON_END_PRINT
    PLAY_MIDI FILE="getitem.mid" C=0
```

**Comment by Tiwatz** [comment]
You machine start and end code should look like this: 

![StartEndGcode](https://github.com/xblax/flashforge_adm5_klipper_mod/assets/162510297/9d13fbe4-988e-4255-8697-3f5443d85a72)

**Reply by Avpman2** [reply]
The extruder_temp command is throwing an error in the console when I set the gcode as:

START_PRINT BED_TEMP=[bed_temperature_initial_layer_single] 
EXTRUDER_TEMP=[nozzle_temperature_initial_layer]

In addition, the nozzle drops down to 200c instead of what I put in the filament settings (265c).

![orca-dev-2](https://github.com/xblax/flashforge_ad5m_klipper_mod/assets/3275462/34b91209-c1b1-4d8b-b308-95833d738c7a)

**Reply by xblax** [reply]
The START_PRINT command with it's parameters must be on a single line, no line break in between.

**Comment by consp** [comment]
You quoted the macro's, you should call those not copy them (like in the image above).

The original Orca macro's are broken as they use M104 and not M109 for the extruder resulting in a "extruder not ready" or "extruder temprature too low" error.

**Comment by Avpman2** [comment]
Found the problem. 💡Instead of removing the old Gcode, I tried commenting it out with a ";". 
In all my years of programming in different languages, I have never come across a language that has TWO types of comments, depending on where they are on a line. The ";" semi-colon in Gcode language is for a comment placed at the END of a line. Therefore:
;G90 
Is NOT commented out. A comment other than at the end of a line in Gcode must be enclosed in ( ). Therefore:
(G90)
is a comment.
G90 ; this is a comment preceded by a semicolon because it's at the END of a line.

Now I have to figure out why my prints don't stick using Orcaslicer but do fine when using Orca-Flashforge. It's the same 3D model with the same settings. (Except the Gcode)

**Reply by DougJoseph** [reply]
Glad you got it sorted. To add another wrinkle to your understanding of comments in G-Code, in some use, the part in parenthesis is not fully ignored, but can be seen by the code parser and used in some way (it's this way with some instances in FluidNC).

Re.
> Now I have to figure out why my prints don't stick using Orcaslicer but do fine when using Orca-Flashforge. It's the same 3D model with the same settings. (Except the Gcode)

The Z height calibration macro process is likely what you need to do. It's not something that needs repeated often unless some change takes place, but it probably needs done after your switch to Klipper tools, at the least.

**Reply by Avpman2** [reply]
How do I do that? And is it for this problem:
![orca](https://github.com/xblax/flashforge_ad5m_klipper_mod/assets/3275462/8f8b40f8-643b-409e-871b-9e70cbc83f8d)

**Reply by DougJoseph** [reply]
@Avpman2 

Re.
> How do I do that? 

If you did install Klipperscreen on the TFT, then tap "More" then "Z Calibrate" then "Start" then "Probe" (as opposed to end stop) — when it comes time for you to manually nudge the rest of the way down to print bed, then use a thin slip of paper and move the nozzle until it is touching decently tightly. You may need to experiment to get the right tension. See screen shots.
   
If you did not install Klipperscreen on the TFT, then you should still have access to the stored value that the above would edit, in one of the config files. Z offset or some such. My saved value is -.195.  

I looked in Mainsail at its list of macros, and did not see a Z calibrate there. 
 
Re. 
> And is it for this problem:

I had that issue at one time that was not related to Z offset but rather to the fact I was trying to print PETG at the temps for PLA. But that must not be your issue because you got the print to work using the other operating system.

Pics of the setup in Klipperscreen

![IMG_7888](https://github.com/xblax/flashforge_ad5m_klipper_mod/assets/8921397/b623f614-b5d6-402e-b74c-2c129331b821)

![IMG_7889](https://github.com/xblax/flashforge_ad5m_klipper_mod/assets/8921397/86a321af-df79-4387-9dff-8a4ca58e129c)

![IMG_7890](https://github.com/xblax/flashforge_ad5m_klipper_mod/assets/8921397/2e32034f-8713-4479-942f-e6e1a18d6eda)

**Comment by Avpman2** [comment]
Drats, I'm running Guppyscreen.
If/when I get the Z-Offset adjusted, will it affect the Orca-Flash version, which is working fine?

**Reply by DougJoseph** [reply]
The Z offset for the Klipper mod should not have any adverse impact whenever you boot into the other operating system. They seem to have their own distinct config files and run independently of each other. At least that’s my understanding.

**Reply by consp** [reply]
No config or system files are shared between the two systems (except some tiny ones to decide which one to boot but those have no impact on the operation of the system after boot).

**Comment by dragos240** [comment]
I'm really confused. I've followed the advice stated here, but when attempting to print, the bed heats up, then the extruder, but immediately afterwards, the extruder stops being heated and has its temperature target cleared. After this, the printer just moves the extruder around and seems confused as to what to do. I really don't know what's up.

**Reply by DougJoseph** [reply]
The only thing I can think of that would cause that, would be having your slicer target a temp of 0 for the project, and I'm guessing you did not do that.

Please share more details of your whole situation.

**Reply by dragos240** [reply]
So I'm using Orcaslicer. I'm assuming that my slicer has some incorrect settings, because if I use raw G-code generated from a [test print website](https://teachingtechyt.github.io/calibration.html#firstlayer), it seems to work. But I'm not sure exactly what settings are incorrect that would lead to this strange behavior. Is it possible for you or anyone else that has Orcaslicer working to share your profile so I can test it out?

**Reply by dragos240** [reply]
It looks like if I set BED_MESH_CALIBRATE as the ABL option, this is what happens. If I turn off that option in that linked website, it starts to print.

**Reply by dragos240** [reply]
Going to "Bed mesh" then calibrating stopped this behavior. From there I was getting it to print, but the outputs were similar to what Avpman2 was experiencing, so I did a Z-calibrate like you instructed. It's doing a test print now

**Reply by dragos240** [reply]
Nope. It's not sticking to the plate even after z-calibration. I'm using PLA with a 55 C bed plate and a 220 C nozzle temperature, which is what I used to get good prints with the previous firmware with Orca-Flashforge. I'm using vanilla OrcaSlicer right now. I'll send over the G-code for the test print I'm doing to see if it's something with the generated G-code soon

EDIT: [test print_PLA.zip](https://github.com/user-attachments/files/16162407/test.print_PLA.zip)

**Reply by DougJoseph** [reply]
> Nope. It's not sticking to the plate even after z-calibration.

In the printer config file is a variable called Z_offset. By default this mod has it set at -0.25. If your first layer is not being laid down close enough to the bed, you can reduce that number. I had to set mine at either -0.19 or -0.15, I can’t remember, in order to get the right amount of squish to get good adhesion. The Z calibrate process is supposed to edit that number, but I think what it actually does is comment out the initial default and put your discovered Z offset into a separate section that is like some important overrides. So take note of whether or not the original Z offset has been commented out, and is being overridden by a new section beneath it. Overrides are denoted by something like #*# or some such at the beginning of each line.

**Reply by dragos240** [reply]
Okay so I FINALLY figured it out. I didn't realize those "commented" lines were affecting the z offset. I manually changed the offset and made sure the "commented" sections didn't have the offset too, and now it works perfectly! Thank you for that final hint!

**Reply by DougJoseph** [reply]
> figured it out

Awesome. Glad it’s working for you!!
